### ISSUE TITLE
HiddenField shows up in the HTML form of the web browsable API page

### ISSUE THREAD

**Issue by lgp171188** [tomchristie/django-rest-framework#2410]
I have a model with a foreign key reference to the User model called 'user'. In my serializer class, which is a sub-class of ModelSerializer, I have overridden the default field created by ModelSerializer and set it to:

```python
user = serializers.HiddenField(default=serializers.CurrentUserDefault())
```

When displaying the fields in the API response, this field is indeed hidden, but in the web browsable API page, the HTML form shows a textbox for the user field which is supposed to be hidden.

Here is my serializer definition:

```python
class CategorySerializer(serializers.ModelSerializer):
    user = serializers.HiddenField(default=serializers.CurrentUserDefault())
    class Meta:
        model = Category
        fields = ('name', 'description', 'created', 'modified','user')
```

Please see the screenshot illustrating the HTML form with an input for the hidden field.

![hiddenfield shown in html form](https://cloud.githubusercontent.com/assets/218870/5726849/a7b53a3a-9b84-11e4-9fba-74e8ac8122ea.png)

I am totally new to Django REST Framework, so sorry if I haven't RTFM and raised this issue.

**Comment by lgp171188** [comment]
I am using version 3.0.3

**Comment by jpadilla** [comment]
Not sure if intentional, but HiddenFields aren't handled in [HTMLFormRenderer](https://github.com/tomchristie/django-rest-framework/blob/master/rest_framework/renderers.py#L339).

**Comment by tomchristie** [comment]
Yup looks valid to me.

**Comment by tomchristie** [comment]
Thanks for raising this.

**Comment by jpadilla** [comment]
@tomchristie submitting a fix...

**Comment by lgp171188** [comment]
This could be misleading because I tried to fill in the username of some other user in the form field to see if I could create a category owned by that user. But it created with the user as the currently logged in user.

**Comment by tomchristie** [comment]
> This could be misleading

Agreed - it is misleading yup.

**Comment by rubendura** [comment]
Is this really fixed? I have updated DRF to 3.0.4 but my hidden fields keep appearing as before for some reason. I've already checked that the changes from the commit closing this issue have been applied, so I have no idea what's wrong with my environment.

**Comment by lgp171188** [comment]
Yes, I can confirm that this fix does not work.

**Comment by jpadilla** [comment]
Closed via #2580
